### PR TITLE
Add an example configurations for using gitlab.com

### PR DESCRIPTION
**Pull request by ableuler**
Add an example configuration for deploying Renku locally against gitlab.com and describe how to use it in the developer docs.

### CODE REVIEW

File: docs/developer/example-configurations/gitlab.rst [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+
+.. _`minikube-values.yaml file`:
+  https://github.com/SwissDataScienceCenter/renku/blob/master/charts/example-configurations/minikube-values-gitlab.yaml
+
+There are two known **caveats** for this setup:
+
+- Notebook servers can only be launched from public projects. This is due to the
+  fact that the users credentials are needed for pulling images for private
+  projects from the gitlab.com registry. These credentials are currently not
+  injected in the init container which does the image pull.
```
Comment (rokroskar) [inline docs/developer/example-configurations/gitlab.rst]:
```suggestion
  injected into the notebook pod.
```

File: docs/developer/example-configurations/gitlab.rst [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+    http://<your-minikube-ip>/auth/realms/Renku/broker/gitlab/endpoint
+    http://<your-minikube-ip>/api/auth/gitlab/token
+    http://<your-minikube-ip>/api/auth/jupyterhub/token
+    http://<your-minikube-ip>/jupyterhub/hub/oauth_callback
+
+
+Step 2
+--------
+Open the file :code:`charts/example-configurations/minikube-values-gitlab.yaml`
+in your cloned Renku repository and paste the Appliation Id and Secret of the
```
Comment (rokroskar) [inline docs/developer/example-configurations/gitlab.rst]:
```suggestion
in your cloned Renku repository and paste the Application Id and Secret of the
```

**Comment by ciyer** [comment]
I was able to use the example and set it up to use renkulab.io. The main difference is in the keycloack setup. Instead of Gitlab, you need to configure an OpenID Connect provider with:
```
Authorization URL: https://renkulab.io/gitlab/oauth/authorize
Token URL: https://renkulab.io/gitlab/oauth/token
```

```Client ID``` and ```Client Secret``` are the ```Application ID``` and ```Secret``` from the Settings > Applications tab in renkulab.io gitlab.

Might make sense to add this example to the documentation as well.

File: charts/example-configurations/minikube-values-gitlab.yaml [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+    postgresPassword: 3b0aa4bae4fbbe167143f583e35eda18
+    sudoToken: 214981ee49479ecbee56359b70ad25dc87edd0e9ea569b7b8c522c3bf979446a
+    urlPrefix: /gitlab
+  keycloak:
+    postgresPassword: 49d0aa1b8a1095415cc975d766933e7b
+  jupyterhub:
+    postgresPassword: d6f59c5df645abe2fd1a0572b6bd5a8d
+  gateway:
+    clientSecret: 259a72d8-2c40-435f-9e0f-24df7c1a7dbc
+    gitlabClientSecret: 4e35410e4116d2280e29c6da6aaaa89fc1ab77364c6352aeb5adbd037d49590a
```
[the remark this answers is not in the document]
Comment (ableuler) [reply charts/example-configurations/minikube-values-gitlab.yaml]: 1. No, it's not - it's the secret of the gateway client application registered in keycloak.
2. No, the client id is actually `renku-ui` - I guess this client is still a relic of adding client applications to gitlab by hacking its database directly. Now the fact that running against gitlab.com actually works suggests that we can get rid of some stuff in the regular setup too.

File: charts/example-configurations/minikube-values-gitlab.yaml [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+    nginx.ingress.kubernetes.io/ssl-redirect: 'false'
+    nginx.ingress.kubernetes.io/proxy-body-size: '0'
+    nginx.ingress.kubernetes.io/proxy-request-buffering: 'off' 
+  hosts:
+  - ""
+
+global:
+  gitlab:
+    clientSecret: e3ecd740-0892-4610-ad69-c7f4886f4533
+    postgresPassword: 3b0aa4bae4fbbe167143f583e35eda18
```
[the remark this answers is not in the document]
Comment (ableuler) [reply charts/example-configurations/minikube-values-gitlab.yaml]: Actually the `sudoToken` should be removed from the regular `minikube-values.yaml` in the first place I would say. It's also true that the `postgresPassword` is obsolete when not running gitlab. But shouldn't this value then be removed from the global section? I would expect the global section to contain values that are used by more than just one service...?

**Comment by ableuler** [comment]
I would suggest to merge this PR and address the issues that are not specific to this PR separately.

File: docs/developer/example-configurations/gitlab.rst [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+
+gitlab.com
+===========
+
+Running GitLab is quite resource-intensive. For development purposes
+or if you already have an existing GitLab instance running, you might want to
+use this pre-existing instance as a backend service for Renku.
+
+Here we describe how to run a local version of the
+platform against gitlab.com **and** also use gitlab.com as identity provider.
```
Comment (rokroskar) [inline docs/developer/example-configurations/gitlab.rst]:
```suggestion
platform against gitlab.com **and** also use gitlab.com as an identity provider.
```

**Comment by rokroskar** [comment]
@ableuler can you please make an issue for the minikube-values cleanup?

**Comment by ableuler** [comment]
> @ableuler can you please make an issue for the minikube-values cleanup?

see #488

File: charts/example-configurations/minikube-values-gitlab.yaml (hunk 4 above)
[the remark this answers is not in the document]
Comment (ableuler) [reply charts/example-configurations/minikube-values-gitlab.yaml]: @leafty ok, thanks for the reminder 👍

**Pull request merged**